src/clients/HubPoolClient.ts

```diff
@@ -646,11 +647,42 @@ export class HubPoolClient {
     this.disputedRootBundles.push(
       ...events["RootBundleDisputed"].map((event) => spreadEventWithBlockNumber(event) as DisputedRootBundle)
     );
-    this.executedRootBundles.push(
-      ...events["RootBundleExecuted"]
-        .filter((event) => !this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber))
-        .map((event) => spreadEventWithBlockNumber(event) as ExecutedRootBundle)
-    );
+
+    const configStoreVersions: { [blockNumber: number]: number } = {};
+    for (const event of events["RootBundleExecuted"]) {
+      if (this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber)) {
+        continue;
+      }
+
+      if (!isDefined(configStoreVersions[event.blockNumber])) {
+        const version = this.configStoreClient.getConfigStoreVersionForBlock(event.blockNumber);
+        configStoreVersions[event.blockNumber] = version;
+      }
+      const version = configStoreVersions[event.blockNumber];
```

Comment:
Good observation from @nicholaspai here: https://github.com/across-protocol/sdk-v2/pull/230#discussion_r1223131959. This PR is using the wrong block number to resolve the version. The correct block number is that of the corresponding root bundle proposal.

Reply:
Should be resolved here: https://github.com/across-protocol/sdk-v2/pull/213/commits/e5023490f9bc62716ccce8b57c9fae9988cdef78.